### ISSUE TITLE
Improve options for `hide` parameter in grc yml

### ISSUE THREAD

**Issue by warnes** [gnuradio/gnuradio#5562]
### Feature Description

The current options for the `hide` option for parameters in grc .yml files seems to be causing confusion.  From https://wiki.gnuradio.org/index.php?title=YAML_GRC#Parameters, the current values are:
> _hide_
> * If the value is **part**, the parameter only be shown in the properties window.
> * If it's **all**, it will never be shown, even inside the properties window.
> * If it's **none**, it will be shown in both the properties window and on the visual block component.

It could improve clarity if these were changed (with aliases for backward compatibility) to:
> _hide_
> * **always** - Do not show, even inside the properties window (was 'all')
> * **never** - Show in both the properties window and on the visual block component (was 'none')
> * **block** - Don't show on the visual block component, but do show in the properties window.

Alternatively, `hide` could be replaced with a `show` parameter, with a reversed meaning, and four options:
> _show_
> * **always** - Show on both the visual block component and in the properties window (default)
> * **properties** - Only show in the properties window
> * **never** - Do not show on either the visual block component nor in the properties window.
> * **block** - Only show on the visual block component.

### Feature Urgency

high (impacts broad portion of ecosystem)

**Comment by haakov** [comment]
I like `always`, `block`, and `never`. I would vote for sticking with `hide`, and keeping the old `all`, `part` and `none` for backward compatibility.

Relevant code:
https://github.com/gnuradio/gnuradio/blob/09c3c4fa4bfb1a02caac74cb5334dfe065391e3b/grc/gui/canvas/block.py#L183

**Comment by dkozel** [comment]
I agree the current hide values and descriptions are not the best for clarity. Changing the values for `hide`, even with aliases, would block new/updated OOTs from being used in current supported and common versions of GNU Radio where they would otherwise work, so I do not think this change should be made in 3.10. There is a lot of discussion around how blocks will work in GNU Radio 4.0+ and it is guaranteed that GR 4.0+ blocks will not work in 3.x versions. This would be a good time to improve and change the parameter/value visibility controls.
